src/hotspot/share/gc/shenandoah/shenandoah_globals.hpp

```diff
@@ -274,12 +292,39 @@
           "reserve/waste is incorrect, at the risk that application "       \
           "runs out of memory too early.")                                  \
                                                                             \
-  product(double, ShenandoahOldEvacReserve, 5.0, EXPERIMENTAL,              \
-          "How much of old generation to withhold from evacuations. "       \
-           "Larger values will result in fewer live objects being "         \
-           "evacuated in the old generation.")                              \
+  product(uintx, ShenandoahOldEvacReserve, 2, EXPERIMENTAL,                 \
+          "How much of old-generation heap to reserve for old-generation "  \
+          "evacuations.  Larger values allow GC to evacuate more live "     \
+          "old-generation objects on every cycle, while potentially "       \
+          "creating greater impact on the cadence at which the young- "     \
+          "generation allocation pool is replenished.  During mixed "       \
+          "evacuations, the bound on amount of old-generation heap "        \
+          "regions included in the collecdtion set is the smaller "         \
+          "of the quantities specified by this parameter and the "          \
+          "size of ShenandoahEvacReserve as adjusted by the value of "      \
+          "ShenandoahOldEvacRatio.  In percents of total old-generation "   \
+          "heap size.")                                                     \
           range(1,100)                                                      \
                                                                             \
+  product(uintx, ShenandoahOldEvacRatioPer128, 16, EXPERIMENTAL,            \
```

Comment:
These `Per128` options are evaluated once per cycle? I would just take the floating point hit for a more user friendly percentage.

Reply:
That's a good improvement.  I'll work on this.

Reply:
This has been addressed.